pkg/controller/certificaterequest/issue_certificate.go

```diff
@@ -36,40 +36,40 @@ import (
 // IssueCertificate validates DNS write access then assess letsencrypt endpoint (prod or stage) based on leclient url.
 // It then iterates through the CertificateRequest.Spec.DnsNames, authorizes to letsencrypt and sets a challenge in the
 // form of resource record. Certificates are then generated and issued to kubernetes via corev1.
-func (r *ReconcileCertificateRequest) IssueCertificate(reqLogger logr.Logger, cr *certmanv1alpha1.CertificateRequest, certificateSecret *corev1.Secret) error {
+func (r *ReconcileCertificateRequest) IssueCertificate(reqLogger logr.Logger, cr *certmanv1alpha1.CertificateRequest, certificateSecret *corev1.Secret) (waitPeriod int, err error) {
```

Comment:
Consider returning a pointer for `waitPeriod` so we're checking `nil` and not `-1`. I'm not entirely sure if that will leak mem though.

Reply:
Yeah, I think returning multiple values is ok here. I'm a little hesitant to add something that would leak memory here.

Reply:
Would like @cblecker opinion here